Search_Engine/Compute/Extract/ExtractAll.cs

```diff
@@ -0,0 +1,103 @@
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using BH.oM.Search;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using FuzzySharp;
+using FuzzySharp.Extractor;
+using FuzzySharp.SimilarityRatio;
+using FuzzySharp.SimilarityRatio.Scorer;
+using FuzzySharp.SimilarityRatio.Scorer.StrategySensitive;
+
+namespace BH.Engine.Search
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Extracts all values unsorted with a score greater than the cutoff when comparing the query to the choices.")]
+        [Input("query", "The string to carry out the fuzzy matching on.")]
+        [Input("choices", "A list of strings to compare the query against.")]
+        [Input("scorer", "The method to use to score the strings when compared.")]
+        [Input("cutOff", "The cuttoff score (i.e. lower bound) for results to be returned.")]
+        [Output("result", "A SearchResult containing the strings, scores and indexes resulting from the fuzzy matching algorithm.")]
+        public static List<SearchResult<string>> ExtractAll(string query, IEnumerable<string> choices, Scorer scorer = Scorer.DefaultRatioScorer, int cutOff = 0)
+        {
+            IRatioScorer scorerMethod = ScorerCache.Get<DefaultRatioScorer>();
+            if (scorer != Scorer.DefaultRatioScorer)
+                scorerMethod = GetScorer(scorer);
+
+            IEnumerable<ExtractedResult<string>> extractedResults = Process.ExtractAll(query, choices.ToArray(), s => s, scorerMethod, cutOff);
+
+            List<SearchResult<string>> results = new List<SearchResult<string>>();
+            foreach (ExtractedResult<string> extractedResult in extractedResults)
+            {
+                SearchResult<string> result = new SearchResult<string>(extractedResult.Value, extractedResult.Score, extractedResult.Index);
+                results.Add(result);
+            }
+
+            return results;
+        }
+
+        /***************************************************/
+
+        [Description("Extracts all BHoMObjects unsorted with a score greater than the cutoff when comparing the query to the choices.")]
+        [Input("query", "The string to carry out the fuzzy matching on.")]
+        [Input("objects", "A list of BHoMObjects to compare the query against.")]
+        [Input("propertyName", "The propertyName to compare the query against - the property must be a string.")]
+        [Input("scorer", "The method to use to score the strings when compared.")]
+        [Input("cutOff", "The cuttoff score (i.e. lower bound) for results to be returned.")]
```

Comment:
```suggestion
        [Input("cutOff", "The cut off score (i.e. lower bound) for results to be returned.")]
```

Reply:
I think cut off is only two words when it is used as a verb: https://www.merriam-webster.com/dictionary/cutoff

Reply:
That's fair, it should only have one t then if we're using the noun then - I'll amend the suggestion

Reply:
Annoyingly used the correct variable name, just fluffed the description :D